storage/perfschema/table_replication_applier_status.cc

```diff
@@ -178,6 +183,14 @@ void table_replication_applier_status::make_row(Master_info *mi)
 
   m_row.count_transactions_retries= mi->rli.retried_trans;
 
+  m_row.log_file_length= static_cast<uint>(strlen(mi->master_log_name));
+  memcpy(m_row.log_file, mi->master_log_name, m_row.log_file_length);
```

Comment:
This is a general note.
We don't have to copy in each invocation. This is a general framework optimization issue though to reference the original string as long the slave applier is alive. Only when it exists the copying would be done.

Reply:
Without understanding the ownership/lifetime rules around dynamically-allocated strings in this part of the code, copy is the only safe way to handle it, right?

> This is a general framework optimization issue though to reference the original string as long the slave applier is alive.

Right. It would also be only a tiny micro-optimization in this case.